hub/integrations/pytorch.py

```diff
@@ -134,13 +160,18 @@ def __getitem__(self, index: int):
             self._all_shared_memory_clean_up()
             self.processed_range = slice(-1, -1)
 
-        return self._apply_transform(sample)
+        if self.tuple_mode:
+            sample = tuple(sample[k] for k in self.keys)
```

Comment:
also, do we really want to allow the user to have both tuple and non-tuple return values? this just seems confusing.

can we instead use `NamedTuple`s? that way you get the best of both worlds? and then you don't need this special casing because you can create a single `NamedTuple` in the constructor (by default it just has all of the tensors)? 

i feel like this will just be more confusing with multiple return types.

Reply:
https://www.geeksforgeeks.org/namedtuple-in-python/

Reply:
I don't think we should use `namedtuple` because namedtuple fields should be valid python identifiers, but tensor names are arbitrary strings. Maybe we can write a custom `OrderedDict` that behaves like a tuple while unpacking but `dict` otherwise. (Python's builtin `OrderedDict` unpacks into keys, but we need values). What do you think?